llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -2848,7 +2848,12 @@ InstructionCost RISCVTargetLowering::getLMULCost(MVT VT) const {
 /// is generally quadratic in the number of vreg implied by LMUL.  Note that
 /// operand (index and possibly mask) are handled separately.
 InstructionCost RISCVTargetLowering::getVRGatherVVCost(MVT VT) const {
-  return getLMULCost(VT) * getLMULCost(VT);
+  auto LMULCost = getLMULCost(VT);
+  if (Subtarget.hasFastVRGather() && LMULCost.isValid()) {
+    unsigned Log = Log2_64(*LMULCost.getValue());
+    return LMULCost * Log;
```

Comment:
Would the cost be zero if LMULCost is 1?

Reply:
Yes, log will return zero if cost is 1, and will return -1 for zero, though `vrgather` is unlikely to have a cost of 1. I can guard this to only produce the number if log value is positive, what do you think?

Reply:
Addressed

Reply:
LGTM